src/server/src/main/java/io/cassandrareaper/jmx/JmxConnectionFactory.java

```diff
@@ -163,4 +175,55 @@ private void decrementSuccessfullConnections(String host) {
       LOG.warn("Could not decrement JMX successfull connections counter for host {}", host, e);
     }
   }
+
+
+
+  public void initializeJmxSeeds(Cluster cluster) {
+    LOG.info("Initializing JMX seed list for cluster {}...", cluster.getName());
+    List<Callable<Optional<String>>> jmxTasks = Lists.newArrayList();
+    List<String> seedHosts = Lists.newArrayList();
+    seedHosts.addAll(cluster.getSeedHosts());
+
+    for (int i = 0; i < seedHosts.size(); i++) {
+      jmxTasks.add(connectToJmx(Arrays.asList(seedHosts.get(i))));
+      if (i % 10 == 0 || i == seedHosts.size() - 1) {
+        tryConnectingToJmxSeeds(jmxTasks);
+        jmxTasks = Lists.newArrayList();
+      }
+    }
+  }
+
+  private Callable<Optional<String>> connectToJmx(List<String> endpoints) {
+    return () -> {
+      try (JmxProxy jmxProxy =
+          this.connectAny(Optional.absent(), endpoints, JmxProxy.JMX_CONNECTION_TIMEOUT)) {
+
+        return Optional.of(endpoints.get(0));
+
+      } catch (RuntimeException e) {
+        LOG.debug("failed to connect to hosts {} through JMX", endpoints.get(0), e);
+        return Optional.absent();
+      }
+    };
+  }
+
+  private void tryConnectingToJmxSeeds(List<Callable<Optional<String>>> jmxTasks) {
+    try {
+      List<Future<Optional<String>>> endpointFutures =
+          initSeedsJmxExecutor.invokeAll(
+              jmxTasks, JmxProxy.JMX_CONNECTION_TIMEOUT, JmxProxy.JMX_CONNECTION_TIMEOUT_UNIT);
+
+      for (Future<Optional<String>> endpointFuture : endpointFutures) {
+        try {
+          endpointFuture.get(JmxProxy.JMX_CONNECTION_TIMEOUT, JmxProxy.JMX_CONNECTION_TIMEOUT_UNIT);
+        } catch (RuntimeException | ExecutionException | TimeoutException e) {
+          // We expect to have failures here, so using TRACE level logs
```

Comment:
if the exception is expected then name the exception `e` as instead `expected` and remove this comment line. that'll explain it enough.

Reply:
Fixed